src/java/fr/paris/lutece/plugins/workflow/modules/appointmentants/service/TaskAntsAppointmentService.java

```diff
@@ -0,0 +1,301 @@
+import fr.paris.lutece.plugins.appointment.service.AppointmentService;
+import fr.paris.lutece.plugins.appointment.service.LocalizationService;
+import fr.paris.lutece.plugins.appointment.web.AppointmentApp;
+import fr.paris.lutece.plugins.appointment.web.dto.AppointmentDTO;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.business.TaskAddAntsAppointmentConfigDAO;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.business.TaskAntsApplicationNumberDAO;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.pojo.AntsStatusResponsePOJO;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.service.rest.TaskAntsAppointmentRest;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.service.rest.TaskAntsAppointmentRestConstants;
+import fr.paris.lutece.portal.service.util.AppLogService;
+import fr.paris.lutece.portal.service.util.AppPropertiesService;
+import fr.paris.lutece.util.httpaccess.HttpAccessException;
+
+/**
+ * 
+ * Class containing useful methods to handle ANTS related tasks
+ * 
+ */
+public class TaskAntsAppointmentService implements ITaskAntsAppointmentService {
+
+	public static final String BEAN_SERVICE = WorkflowAppointmentAntsPlugin.PLUGIN_NAME + ".taskAntsAppointmentService";
+
+	@Inject
+	@Named( TaskAntsApplicationNumberDAO.BEAN_NAME )
+	private TaskAntsApplicationNumberDAO _application_number_dao;
+
+	@Inject
+	@Named( TaskAddAntsAppointmentConfigDAO.BEAN_NAME )
+	private TaskAddAntsAppointmentConfigDAO _task_add_appointment_dao;	
+
+	/**
+	 * Variables for general use
+	 */
+	public static final String KEY_URL = "url";
+	public static final String KEY_LOCATION = "location";
+	public static final String KEY_DATE = "date";	
+
+	private TaskAntsAppointmentService( )
+	{
+	}
+
+	/**
+	 * Check if an appointment was created from the front office or from the back office
+	 * @param appointment the appointment to check
+	 * @return true if it was created by a user in the front office, returns false otherwise
+	 */
+	public static boolean isAppointmentCreatedInFrontOffice( Appointment appointment )
+	{
+		/* If the appointment's "AdminUserCreate" field has no value OR has a value different from
+		 * the Admin code value ('admin' by default), then it was created by a user in the front office 
+		 * */
+		String adminCode = AppPropertiesService.getProperty( "appointment.ants.admin.code.value" );
+		return appointment.getAdminUserCreate( ) == null ||
+				appointment.getAdminUserCreate( ).isEmpty( ) ||
+				!appointment.getAdminUserCreate( ).equalsIgnoreCase( adminCode );
+	}
+
+	public static String createAntsAddAppointmentUrl( String baseUrl, String addAppointmentUrl, String applicationId,
+			String managementUrl, String meetingPoint, String dateTime )
+	{
+		return buildAntsAddAppointmentUrl( baseUrl, addAppointmentUrl,
+				AppPropertiesService.getProperty( TaskAntsAppointmentRestConstants.ANTS_APPLICATION_ID ) + '=' + applicationId,
```

Comment:
I appreciate your consideration of loading the properties during class loading using the classloader, rather than with each execution of the task.